src/site/content/en/blog/carousel-best-practices/index.md

```diff
@@ -0,0 +1,170 @@
+
+    A surprising number of carousels use janky, non-composited animations that can contribute to CLS. On pages with autoplaying carousels, this has the potential to cause infinite CLS. This  type of CLS typically isn't apparent to the human eye, which makes this issue easy to overlook. To avoid this issue, [avoid using non-composited animations](https://web.dev/non-composited-animations/) in your carousel (for example, during slide transitions).
+
+
+### Best practices
+
+Optimizing the performance of a carousel entails optimizing both its technical architecture and its content.
+
+
+#### Load carousel content using HTML
+
+Carousel content should be loaded via the page's HTML markup so that it is discoverable by the browser early in the page load process. Using JavaScript to initiate the loading of carousel content is probably the single biggest performance mistake to avoid when using carousels. This delays image loading and can negatively impact LCP.
+
+An advanced carousel optimization that deviates slightly from this approach consists of loading the first slide statically, then progressively enhancing it to include navigation controls and additional content. This technique is most applicable to environments where you have a user's prolonged attention - this gives the additional content time to load. In environments like home pages, where users may only stick around for a second or two, only loading a single image may be similarly effective.
+
+
+#### Use modern technology
+
+Many sites use third-party JavaScript libraries to implement carousels. If you currently use older carousel tooling, you may be able to improve performance by switching to newer tooling. Newer tools tend to use more efficient APIs and are less likely to require additional dependencies like jQuery.
+
+However, you may not need JavaScript at all. The new [scroll-snap](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Scroll_Snap) API makes it possible to build full-featured, native-like carousels using only HTML and CSS - no JavaScript required.
+
+Here are some resources on using scroll-snap that you may find helpful:
+
+
+
+*   [Building a Stories component (web.dev)](https://web.dev/building-a-stories-component/)
+*   [Next-generation web styling: scroll snap (web.dev)](https://web.dev/next-gen-css-2019/#scroll-snap)
+*   [CSS-Only Carousel (CSS Tricks)](https://css-tricks.com/css-only-carousel/)
+*   [How to Make a CSS-Only Carousel (CSS Tricks)](https://css-tricks.com/how-to-make-a-css-only-carousel/)
+
+
+#### Optimize carousel content
+
+Carousels often contain some of a site's largest images.  Thus, it can be worth your time to make sure that these images are fully optimized. Choosing the right image format and compression level, [using an image CDN](https://web.dev/image-cdns), and [using srcset to serve multiple image versions](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Scroll_Snap) are all techniques that can reduce the transfer size of images.
```

Comment:
```suggestion
Carousels often contain some of a site's largest images, so it can be worth your time to make sure that these images are fully optimized. Choosing the right image format and compression level, [using an image CDN](/image-cdns), and [using srcset to serve multiple image versions](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Scroll_Snap) are all techniques that can reduce the transfer size of images.
```